clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -167,7 +167,7 @@
  * called <i>test</i> as described above.
  * <p>
  * The broker will automatically detect failed processes in the <i>test</i> group by using a heartbeat mechanism. The
- * consumer will automatically ping the cluster periodically, which lets the cluster know that it is alive. As long as
+ * consumer will ping the cluster whenever {@link #poll(long)} is called, which lets the cluster know that it is alive. As long as
```

Comment:
Yeah, that sounds good to me. One minor suggestion in the last sentence.

> If it stops heartbeating **by failing** to call poll() for a period of time longer than session.timeout.ms then it will be considered dead and its partitions will be assigned to another process.

Does that make sense?